libraries/helpers.rb

```diff
@@ -31,6 +31,29 @@ def systemd_is_pid_1?
 
     module_function :module_loaded?, :systemd_is_pid_1?
   end
+
+  # Create a stub class for evaluating resource sub-blocks outside resource context
+  class OptionEvalContext
+    # parent is the parent resource
+    # context is the sub-block we're in
+    def initialize(parent, context)
+      @parent = parent
+      @context = context
+    end
+
+    def method_missing(method_sym, *args, &block)
+      target_meth = "#{@context}_#{method_sym}".to_sym
```

Comment:
meth: not even once

Reply:
![T.hanks](http://lolsheaven.com/wp-content/uploads/2012/12/Thanks-Tom-Hanks.jpg)